.gitlab-ci.yml

```diff
@@ -0,0 +1,62 @@
+stages:
+  - deploy_to_sonatype
+  - create_key
+
+variables:
+  REGISTRY: 486234852809.dkr.ecr.us-east-1.amazonaws.com
+
+deploy_to_sonatype:
+  stage: deploy_to_sonatype
+  tags: [ "runner:docker", "size:large" ]
+  image: maven:3.8.2-jdk-8-slim
+  when: manual
+
+  script:
+    # Ensure we don't print commands being run to the logs during credential
+    # operations
+    - set +x
+
+    - apt update
+    - apt install -y python3 python3-pip
+    - python3 -m pip install awscli
+
+    - echo "Fetching Sonatype user..."
+    - export SONATYPE_USER=$(aws ssm get-parameter --region us-east-1 --name ci.jmeter-datadog-backend-listener.publishing.sonatype_username --with-decryption --query "Parameter.Value" --out text)
+
+    - echo "Fetching Sonatype password..."
+    - export SONATYPE_PASS=$(aws ssm get-parameter --region us-east-1 --name ci.jmeter-datadog-backend-listener.publishing.sonatype_password --with-decryption --query "Parameter.Value" --out text)
+
+    - echo "Fetching signing key password..."
+    - export GPG_PASSPHRASE=$(aws ssm get-parameter --region us-east-1 --name ci.jmeter-datadog-backend-listener.signing.gpg_passphrase --with-decryption --query "Parameter.Value" --out text)
+
+    - echo "Fetching signing key..."
+    - GPG_KEY=$(aws ssm get-parameter --region us-east-1 --name ci.jmeter-datadog-backend-listener.signing.gpg_private_key --with-decryption --query "Parameter.Value" --out text)
+
+    - printf -- "$GPG_KEY" | gpg --import --batch
+
+    - set -x
+
+    - echo "Building release..."
+    - mvn -DperformRelease=true -DskipTests -Darguments=-DskipTests --settings ./settings.xml clean deploy
+    - echo "Release built"
+
+# This job creates the GPG key
+# NOTE: This is included for once gpg keys expire, this has not been ran before
+create_key:
+  stage: create_key
+  when: manual
+
+  tags:
+    - "runner:docker"
+    - "size:large"
+
+  image: $REGISTRY/ci/agent-key-management-tools/gpg:1
+  variables:
+    PROJECT_NAME: "jmeter-datadog-backend-listener"
+    EXPORT_TO_KEYSERVER: "false"
+  script:
+    - /create.sh
```

Comment:
Where does this come from?

Reply:
The `create_key` stage was copied from: https://github.com/DataDog/agent-key-management-tools/tree/master/gpg. The `create.sh` script itself comes from https://github.com/DataDog/agent-key-management-tools/blob/master/gpg/create.sh, although I haven't ran this script yet since the keys are already valid and didn't need to be changed.

Reply:
So we're planning to reuse the key? That's an important consideration that needs to be noted somewhere.

Reply:
Yes, the keys are reused.